src/audio/drc/drc_math_generic.c

```diff
@@ -113,7 +113,7 @@ inline int32_t drc_sin_fixed(int32_t x)
 	const int32_t PI_OVER_TWO = Q_CONVERT_FLOAT(1.57079632679489661923f, 30);
 
 	/* input range of sin_fixed() is non-negative */
-	int32_t abs_sin_val = sin_fixed(q_mult(ABS(x), PI_OVER_TWO, 30, 30, 28));
+	int32_t abs_sin_val = sin_fixed(q_mult(ABS(x), PI_OVER_TWO, 30, 30, 29));
```

Comment:
i thought only the output was changed? why are we changing the input

Reply:
Yeah it should not be changed.
And if you change the output q-format for sin_fixed(), you must take care of all existing callers to avoid errors.
I think you should add:
abs_sin_val = sat_int32(Q_SHIFT_LEFT((int64_t)abs_sin_val, 30, 31));

Reply:
OK - I will modify accordingly